### DISCUSSION TITLE
Can files/folders be created in a loop?

### DISCUSSION THREAD

**Discussion by zedlander1000** [copier-org/copier#567] [Q&A]
Say I have a question:
```
Enter a list of directory names:
input: ['dir1', 'dir2', 'dir3']
```
How could I loop through that list and create a directory for each name?
For example:
I have the `src` directory and within the `src` directory, I want to create `dir1`, `dir2`, and `dir3`.

```jinja
[% for dir in dir_names %][[dir]][% endfor %]
```
The above jinja simply just concatenates the names into a single directory.
There either is some mechanism i am not aware of, or this simply is not possible.

If this is not possible, this would be a wonderful feature to have, though I assume there are many design questions that need to be answered in order to make that work. Otherwise, would this be possible with jinja extensions?

**Comment by pawamoy** [comment]
That's indeed not possible right now. It would be complicated to design and implement I think. You would typically use [tasks](https://copier.readthedocs.io/en/latest/configuring/#tasks) to handle this use-case.

**Reply by zedlander1000** [reply]
Could copier (or its users with a custom jinja extension) potentially create a custom Jinja tag to handle this?
i.e.
```
[% createfor <dir or file name> in <list of names> %]...[% endcreatefor %]
```
I'm sure it is probably a lot less simple than adding a custom tag since the loop would have to respect subdirs and child files.

**Reply by yajo** [reply]
Possibly you'll be able to achieve what you want with a bit of creativity and the proposed solution of tasks.

If so, please share around here, to help others 😃

**Comment by pawamoy** [comment]
See #1271 btw.
